package com.protos;

option java_package = "com.protos";
option java_outer_classname = "PlayerModelProto";

import "deck.proto";
import "hero.proto";
import "minion.proto";
import "card.proto";

message PlayerModel {
  required string name = 1;
  required int32 player_id = 2;
  required Hero hero = 3;
  required Deck deck = 4;
  optional int32 mana = 5;
  required int32 max_mana = 6;
  optional int32 deck_pos = 7;
  optional int32 fatigue_damage = 8;
  repeated Minion minions = 9;
  repeated Card hand = 10;
  optional int32 overload = 11;
  optional int32 num_cards_used = 12;
}